port/common/omrsysinfo_helpers.c

```diff
@@ -433,16 +498,27 @@ omrsysinfo_get_x86_description(struct OMRPortLibrary *portLibrary, OMRProcessorD
 	desc->features[1] = CPUInfo[CPUID_ECX];
 	desc->features[2] = 0;
 
-	/* If OSXSAVE is supported, populate the XSAVE state in desc->features[2]. Unused bits (4-31) are reserved for future expansion. */
+	/* If OSXSAVE is supported, populate the XSAVE state in desc->features[2], bits 0-3 */
 	if (OMR_ARE_ANY_BITS_SET(desc->features[(OMR_FEATURE_X86_OSXSAVE) / 32], INDEX_TO_MASK(OMR_FEATURE_X86_OSXSAVE))) {
 		desc->features[2] |= omrsysinfo_get_x86_xsave_state();
 	}
 
 	/* extended features */
-	omrsysinfo_get_x86_cpuid_ext(CPUID_STRUCTURED_EXTENDED_FEATURE_INFO, 0, CPUInfo); /* 0x0 is the only valid subleaf value for this leaf */
+	omrsysinfo_get_x86_cpuid_ext(CPUID_STRUCTURED_EXTENDED_FEATURE_INFO, 0, CPUInfo); /* CPUID.(0x7.0x0) */
 	desc->features[3] = CPUInfo[CPUID_EBX]; /* Structured Extended Feature Flags in EBX */
 	desc->features[4] = CPUInfo[CPUID_ECX]; /* Structured Extended Feature Flags in ECX */
 
+	omrsysinfo_get_x86_cpuid_ext(CPUID_STRUCTURED_EXTENDED_FEATURE_INFO, 1, CPUInfo); /* CPUID.(0x7.0x1). */
+	desc->features[5] = CPUInfo[CPUID_EAX]; /* Structured Extended Feature Flags in EAX */
+	desc->features[6] = CPUInfo[CPUID_EDX]; /* Structured Extended Feature Flags in EDX */
+
+	/* If AVX 10 is supported, populate the AVX 10.X sub version flags in desc->features[2], bits 4-6, 16-18
+	 * Unused bits (7-15, 19-31) are reserved for future expansion.
+	 */
+	if (OMR_ARE_ANY_BITS_SET(desc->features[(OMR_FEATURE_X86_AVX10) / 32], INDEX_TO_MASK(OMR_FEATURE_X86_AVX10))) {
+		desc->features[2] |= omrsysinfo_get_x86_avx10_subversion();
+	}
```

Comment:
all code that modifies `desc->features[2]` should stay together; we can move this to the vicinity of line 505